subsys/bluetooth/gatt_dm.c

```diff
@@ -61,69 +70,97 @@ struct bt_gatt_dm {
 /* Currently only one instance is supported */
 static struct bt_gatt_dm bt_gatt_dm_inst;
 
-
-static void *user_data_store(struct bt_gatt_dm *dm,
-			     const void *user_data,
+/* Returns pointer to newly allocated space in a dm->data_chunk */
+static void *user_data_alloc(struct bt_gatt_dm *dm,
 			     size_t len)
 {
 	u8_t *user_data_loc;
-	size_t cur_chunk_id = dm->data_chunk.cur_id;
+	struct data_chunk_item *new_item;
 
-	__ASSERT_NO_MSG(len <= CHUNK_SIZE);
+	/* Round up len to 32 bits to make sure that return pointers are always
+	 * correctly aligned.
+	 */
+	len = (len + DATA_ALIGN - 1) & ~(DATA_ALIGN - 1);
 
-	if (!(dm->data_chunk.chunks[cur_chunk_id])) {
-		dm->data_chunk.chunks[cur_chunk_id] = k_malloc(CHUNK_SIZE);
-		dm->data_chunk.cur_len = 0;
+	__ASSERT_NO_MSG(len <= CHUNK_DATA_SIZE);
 
-		if (!(dm->data_chunk.chunks[cur_chunk_id])) {
-			return NULL;
-		}
-	}
-
-	size_t misalign = dm->data_chunk.cur_len % UUID_ALIGN;
-	if (misalign) {
-		dm->data_chunk.cur_len += UUID_ALIGN - misalign;
-	}
+	if (!dm->data_chunk.cur
+			|| dm->data_chunk.cur_len + len > CHUNK_DATA_SIZE) {
```

Comment:
coding standard: Move || up to preceding line and left-align this line so that has same identation as the ! of the above line.

Reply:
I will fix that. I wasn't aware that coding standard says something about that. I am familiar with following two documents, are there some more documents regarding coding standard?
https://docs.zephyrproject.org/latest/contribute/index.html#coding-style
https://www.kernel.org/doc/html/latest/process/coding-style.html

Reply:
Maybe it's not there. But it is consistent with the rest of the code. :)